gamemode/modules/base/sv_gamemode_functions.lua

```diff
@@ -263,9 +275,10 @@ function GM:OnNPCKilled(victim, ent, weapon)
     end
 
     -- If we know by now who killed the NPC, pay them.
-    if IsValid(ent) and GAMEMODE.Config.npckillpay > 0 then
-        ent:addMoney(GAMEMODE.Config.npckillpay)
-        DarkRP.notify(ent, 0, 4, DarkRP.getPhrase("npc_killpay", DarkRP.formatMoney(GAMEMODE.Config.npckillpay)))
+    if IsValid(ent) and hook.Call("canEarnNPCMoney", GAMEMODE, ent, victim) then
+        local amount = hook.Call("calculateNPCMoney", GAMEMODE, ent, victim)
```

Comment:
From my point of view, I think this would be more descriptive if named calculateNPCKillAward or calculateNPCKillPay or similar. To me, "NPC money" is quite vague - it could be how much money the NPC is holding, how much one costs to spawn, ...

Maybe it's just me though.